Allow linked data definitions to be rendered intuitively in the `additional_display` tabs

jsheunis opened the pull request:
This applies changes from [datalad-catalog 30773cb8797c39cf963f3a147cc51b9f79df0056](https://github.com/datalad/datalad-catalog/commit/30773cb8797c39cf963f3a147cc51b9f79df0056) which allows linked data definitions to be rendered intuitively in the additional display tabs.

It addresses https://github.com/psychoinformatics-de/sfb1451-projects-catalog/issues/41 in part, since it only applies to the `additional_display` tab for SFB1451, and not for fields rendered on the main dataset page or other tabs (content, subdatasets, etc).

jsheunis committed:
apply changes from datalad-catalog 30773cb8797c39cf963f3a147cc51b9f79df0056 which allows linked data definitions to be rendered intuitively in the additional display tabs

mslw commented:
This seems to entirely break the display of "regular" datasets (i.e. those not coming from tabby), including the superdataset. This is the error from Firefox console:

```
TypeError: disp_dataset.additional_tabs is undefined
    dataset_ready http://localhost:8000/assets/app_component_dataset.js:208
    Be http://localhost:8000/assets/vue.2.6.14.min.js:6
    run http://localhost:8000/assets/vue.2.6.14.min.js:6
    ln http://localhost:8000/assets/vue.2.6.14.min.js:6
    Qe http://localhost:8000/assets/vue.2.6.14.min.js:6
    We http://localhost:8000/assets/vue.2.6.14.min.js:6
    promise callback*ze http://localhost:8000/assets/vue.2.6.14.min.js:6
    Qe http://localhost:8000/assets/vue.2.6.14.min.js:6
    update http://localhost:8000/assets/vue.2.6.14.min.js:6
    update http://localhost:8000/assets/vue.2.6.14.min.js:6
    notify http://localhost:8000/assets/vue.2.6.14.min.js:6
    set http://localhost:8000/assets/vue.2.6.14.min.js:6
    set http://localhost:8000/assets/vue.2.6.14.min.js:6
    created http://localhost:8000/assets/app_component_dataset.js:711
    Be http://localhost:8000/assets/vue.2.6.14.min.js:6
    Qt http://localhost:8000/assets/vue.2.6.14.min.js:6
    _init http://localhost:8000/assets/vue.2.6.14.min.js:6
    a http://localhost:8000/assets/vue.2.6.14.min.js:6
    componentInstance http://localhost:8000/assets/vue.2.6.14.min.js:6
    init http://localhost:8000/assets/vue.2.6.14.min.js:6
    n http://localhost:8000/assets/vue.2.6.14.min.js:6
    f http://localhost:8000/assets/vue.2.6.14.min.js:6
    f http://localhost:8000/assets/vue.2.6.14.min.js:6
    x http://localhost:8000/assets/vue.2.6.14.min.js:6
    x http://localhost:8000/assets/vue.2.6.14.min.js:6
    x http://localhost:8000/assets/vue.2.6.14.min.js:6
    x http://localhost:8000/assets/vue.2.6.14.min.js:6
    zi http://localhost:8000/assets/vue.2.6.14.min.js:6
    _update http://localhost:8000/assets/vue.2.6.14.min.js:6
    r http://localhost:8000/assets/vue.2.6.14.min.js:6
    get http://localhost:8000/assets/vue.2.6.14.min.js:6
    run http://localhost:8000/assets/vue.2.6.14.min.js:6
    ln http://localhost:8000/assets/vue.2.6.14.min.js:6
    Qe http://localhost:8000/assets/vue.2.6.14.min.js:6
    We http://localhost:8000/assets/vue.2.6.14.min.js:6
    promise callback*ze http://localhost:8000/assets/vue.2.6.14.min.js:6
    Qe http://localhost:8000/assets/vue.2.6.14.min.js:6
    update http://localhost:8000/assets/vue.2.6.14.min.js:6
    update http://localhost:8000/assets/vue.2.6.14.min.js:6
    notify http://localhost:8000/assets/vue.2.6.14.min.js:6
    set http://localhost:8000/assets/vue.2.6.14.min.js:6
    init http://localhost:8000/assets/vue-router.3.5.3.min.js:11
    init http://localhost:8000/assets/vue-router.3.5.3.min.js:11
```

It looks like something I should be able to fix. I'll dig in.

mslw committed:
Fix dataset display in absence of additional tabs

This fixes JS "TypeError: disp_dataset.additional_tabs is undefined"
which prevented datasets without additional_tabs from displaying. We
now explicitly check whether the additional_tabs is an array before
checking its length.

mslw reviewed (approved):
With a small tweak this seems to work well. Datasets without `additional_tab_defs` are displayed as previously, while datasets with that information have the affected tab rendered with links. What's more, it seems it will be easy to revert the changes whenever we decide to go for a different implementation.

jsheunis commented:
Nice catch! I agree, this is a sensible fix and easy to revert if we figure out an improved way in future.
